src/ConfigFileParser.cpp

```diff
@@ -0,0 +1,534 @@
+        if (value.find('.') == std::string::npos)
+            throw std::runtime_error("Invalid port");
+        ip = num;
+    } else
+        ip = value.substr(0, colonIndex);
+
+    if (!isIpAddressValid(ip))
+        throw std::runtime_error("Invalid ip address");
+
+    m_configFile.servers[m_configFile.serverIndex].listen.insert(std::make_pair(ip, 0));
+}
+
+/**
+ * @brief Checks if the value of the listen directive is valid and reads it if that is the case
+ *
+ * The function makes sure that the port is valid in the following ways:
+ * 1. The port must not contain a character other than '0'-'9'
+ * 2. The value of the port must be between 1-65535
+ *
+ * @param port The port to be checked
+ * @return true If the port is valid
+ * @return false If the port is invalid
+ */
+bool ConfigFileParser::isPortValid(const std::string& port) const
+{
+    if (port.find_first_not_of("0123456789") != std::string::npos)
+        return false;
+
+    const int base = 10;
+    const int maxPort = 65535;
+    const int minPort = 1;
+
+    int long portNum = std::strtol(port.c_str(), NULL, base);
+    return !(portNum <= minPort || portNum > maxPort);
+}
+
+/**
+ * @brief Checks if the port is valid and reads it if that is the case
+ *
+ * For the case that no semicolon is can be found in the listen directive, the function checks
+ * if the defined values is valid ip address. If that is true, the function returns.
+ * If that is not true, the function and checks if it is a valid port and reads it.
+ *
+ */
+void ConfigFileParser::readPort(const std::string& value)
+{
+    size_t colonIndex = value.find(':');
+    size_t semicolonIndex = value.find(';');
+    std::string port;
+
+    if (colonIndex == std::string::npos) {
+        std::string num = value.substr(0, semicolonIndex);
+        if (isIpAddressValid(num))
+            return;
+        if (value.find('.') != std::string::npos)
+            throw std::runtime_error("Invalid ip address");
+        port = num;
+    } else
+        port = value.substr(colonIndex + 1, semicolonIndex - colonIndex - 1);
+
+    if (!isPortValid(port))
+        throw std::runtime_error("Invalid port");
+
+    for (std::map<std::string, unsigned short>::iterator it = m_configFile.servers[m_configFile.serverIndex].listen.begin(); it != m_configFile.servers[m_configFile.serverIndex].listen.end(); it++)
+        if (it->second == 0)
+            it->second = std::atoi(port.c_str());
+}
+
+/**
+ * @brief Reads the root path
+ *
+ * The function checks if the root path is valid and reads it if that is the case.
+ *
+ * It makes sure that the path is valid in the following ways:
+ * 1. There is only one root path
+ *
+ * If at the end of the path is a slash, it removes it.
+ *
+ * This function can be used for the server block and location block
+ *
+ * @param block The block which surounds the directive
+ */
+void ConfigFileParser::readRootPath(int block, const std::string& value)
+{
+    std::string rootPath = value;
```

Comment:
I wanted to keep it consistent with the other functions. 

Furthermore the rootPath var is needed for storing the substring of the root path in case of an slash at the end. 
This would not work because the param is const:

```cpp
if (rootPath[rootPath.length() - 1] == '/')
		rootPath = rootPath.substr(0, rootPath.length() - 1);
```

Reply:
True, I missed that!

Another bit I just saw: `int block` could be the enum, so you don't have to convert

Reply:
Absolutely. Changed the datatype to Block instead of int for all corresponding functions